pylana/aggregation/core.py

```diff
@@ -0,0 +1,227 @@
+            activities:
+                An optional list denoting the activities to use when grouping
+                is set to 'byActivity'.
+            date_type:
+                An optional string denoting the date type to use when a time
+                grouping is used. It has to be 'startDate' or 'endDate'.
+            secondary_attribute:
+                An optional string denoting the attribute to use when secondary
+                grouping is set to 'byAttribute'.
+            secondary_activities:
+                An optional list denoting the activities to use when secondary
+                grouping is set to 'byActivity'.
+            secondary_date_type:
+                An optional string denoting the date type to use when a secondary
+                time grouping is used. It has to be 'startDate' or 'endDate'.
+            **kwargs:
+                Keyword arguments passed to requests functions.
+
+        Returns:
+            A pandas DataFrame containing the aggregated data.
+        """
+        request_data = {'metric': create_metric(metric, aggregation_function, percentile),
+                        'valuesFrom': {'type': values_from},
+                        'miningRequest': {'logId': log_id,
+                                          'activityExclusionFilter': activity_exclusion_filter,
+                                          'traceFilterSequence': trace_filter_sequence},
+                        'options': {'maxAmountAttributes': max_amount_attributes,
+                                    'valueSorting': value_sorting,
+                                    'sortingOrder': sorting_order}}
+
+        if grouping is not None:
+            request_data['grouping'] = create_grouping(grouping, date_type, activities,
+                                                       attribute)
+
+        if secondary_grouping is not None:
+            request_data['secondaryGrouping'] = create_grouping(secondary_grouping,
+                                                                secondary_date_type,
+                                                                secondary_activities,
+                                                                secondary_attribute
+                                                                )
+
+        aggregate_response = self.post('/api/v2/aggregate-data', json=request_data, **kwargs)
+
+        if aggregate_response.status_code >= 400:
+            return pd.DataFrame()
+
+        response_df = extract_chart_values(aggregate_response.json())
+        if secondary_grouping is not None:
+            response_df = normalise_chart_values(response_df, 'values')
+
+        response_df = response_df.rename(columns={'xAxis': attribute if attribute is not None else grouping,
+                                                  'yAxis': metric,
+                                                  'zAxis': secondary_attribute if secondary_attribute is not None else secondary_grouping})
```

Comment:
This can be styled to shorter lines. Google Styleguide and PEP 8 recommend 80 characters, with some exceptions.